chore(ci): add workflow for scanning unicorn core for CVEs

mjnagel opened the pull request:
## Description

This adds a new task file for scanning for CVEs as well as a workflow to run a nightly scan against the latest unicorn flavor core release.

The tasks are built to be dynamic to allow for scanning other packages/versions/flavors as needed and adjust the reporting threshold.

The workflow will specifically scan the latest release of core (unicorn flavor) and create a github issue with all High/Critical CVEs listed for triaging/fixing.

## Related Issue

Fixes https://github.com/defenseunicorns/uds-core/issues/1273

Closes https://github.com/defenseunicorns/uds-core/pull/1162 - this will replace the need to trigger security-hub's scanning.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Steps to Validate

This workflow was partially tested on my test repo here (run against registry1 flavor due to permissions for private packages):
- CI Run (note artifact for the scan): https://github.com/mjnagel/test/actions/runs/13297486419
- Open Dashboard Issue (note that this has been edited by second run): https://github.com/mjnagel/test/issues/106

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

mjnagel committed:
chore(ci): add workflow for scanning unicorn core for CVEs

mjnagel committed:
chore: add grype install

mjnagel committed:
fix: permissions for workflow

mjnagel committed:
fix: grype-markdown.tmpl

mjnagel commented:
Going to make a few changes here:
- Updating this to run right after a release so that we have a fresh scan immediately
- May update the format of the output to be more readable/concise

koesbong commented:
> Going to make a few changes here:
> 
> * Updating this to run right after a release so that we have a fresh scan immediately
> * May update the format of the output to be more readable

So it'll run nightly and also right after a release?

mjnagel commented:
@koesbong correct yeah, that way a scan is never older than 24 hours, but we also have a scan immediately upon release.

mjnagel committed:
chore: add doc, cleanup format, run on release

mjnagel committed:
Merge branch 'main' into cve-scan-workflow

mjnagel committed:
Merge branch 'main' into cve-scan-workflow

mjnagel commented:
Updated, workflow will now run everyday + right after a release is published. The format was also updated for the markdown to be more concise and use dropdowns for each image (hiding the full CVE tables by default). Example of that can be viewed on the [linked github issue](https://github.com/mjnagel/test/issues/106) or from the artifact for the [most recent job here](https://github.com/defenseunicorns/uds-core/actions/runs/13313953778). I also added a basic task readme with an example of how to run this scan job locally.

chance-coleman reviewed (approved):
LGTM
